routing/single_mwm_router.cpp

```diff
@@ -264,7 +264,8 @@ bool SingleMwmRouter::BuildRoute(MwmSet::MwmId const & mwmId, vector<Joint::Id>
   shared_ptr<traffic::TrafficInfo::Coloring> trafficColoring = m_trafficCache.GetTrafficInfo(mwmId);
 
   auto const numJunctions = junctions.size();
-  ReconstructRoute(m_directionsEngine.get(), m_roadGraph, trafficColoring, delegate, start, finish,
+  CHECK(m_directionsEngine, ());
```

Comment:
I'd prefer to use CHECK in such cases to be sure that the app is terminated and write to log the line before * m_directionsEngine.